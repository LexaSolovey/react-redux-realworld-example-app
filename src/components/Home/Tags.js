import React from 'react';

const Tags = ({tags, fetchArticlesByTag}) => {
  const callFetchAtricles = (event) => {
    event.preventDefault();

    const {tagName, attributes} = event.target;

    if(tagName === 'A' && attributes['data-tagname'].value) {
      fetchArticlesByTag(attributes['data-tagname'].value)
    }
  };

  if (tags) {
    return (
      <div className="tag-list" onClick={callFetchAtricles}>
        {tags.map((tag) => (
            <a
              href=""
              className="tag-default tag-pill"
              key={tag}
              data-tagname={tag}>
              {tag}
            </a>
          )
        )}
      </div>
    );
  } else {
    return (
      <div>Loading Tags...</div>
    );
  }
};

export default Tags;
