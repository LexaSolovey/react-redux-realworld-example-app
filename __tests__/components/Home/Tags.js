import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { expect as chaiExpext } from 'chai';

import { tagsList } from '../../data/tagsList'; 
import Tags from '../../../src/components/Home/Tags';

const fetchArticlesByTag = jest.fn();

describe('>> Tags component', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Tags tags={tagsList} fetchArticlesByTag={fetchArticlesByTag} />);
    });

    it('snapshot returns null in case when tags is not exist', () => {
        const wrapper = renderer.create(<Tags />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
    
    it('snapshot renders tagsList', () => {
        const wrapper = renderer.create(<Tags tags={tagsList} />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correct number of items', () => {
        chaiExpext(wrapper.find('.tag-default.tag-pill')).to.have.lengthOf(tagsList.length);
    });

    it('callFetchAtricles condition ckeck. fetchArticlesByTag should not be called', () => {
        const event = {
            preventDefault: () => {},
            target: {
                tagName: 'A',
                attributes: {
                    'data-tagname': {}
                }
            }
        };
        
        wrapper.find('div.tag-list').simulate('click', event);
        expect(fetchArticlesByTag).not.toHaveBeenCalled();
    });

    it('callFetchAtricles condition ckeck. fetchArticlesByTag should be called', () => {
        const event = {
            preventDefault: () => {},
            target: {
                tagName: 'A',
                attributes: {
                    'data-tagname': {
                        value: 'butt'
                    }
                }
            }
        };
        
        wrapper.find('div.tag-list').simulate('click', event);
        expect(fetchArticlesByTag).toHaveBeenCalled();
    });

    it('handleClick executes preventDefault()', () => {
        let prevented = false; 
        wrapper.find('div.tag-list')
            .simulate('click', {
                preventDefault: () => {
                    prevented = true;
                },
                target: {
                    tagName: 'A',
                    attributes: {
                        'data-tagname': {
                            value: 'butt'
                        }
                    }
                }
            });

        expect(prevented).toBe(true);
    });
})
