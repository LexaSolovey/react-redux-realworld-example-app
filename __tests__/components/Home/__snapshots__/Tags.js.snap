// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>> Tags component snapshot renders tagsList 1`] = `
<div
  className="tag-list"
  onClick={[Function]}
>
  <a
    className="tag-default tag-pill"
    data-tagname="butt"
    href=""
  >
    butt
  </a>
  <a
    className="tag-default tag-pill"
    data-tagname="dragons"
    href=""
  >
    dragons
  </a>
  <a
    className="tag-default tag-pill"
    data-tagname="test"
    href=""
  >
    test
  </a>
  <a
    className="tag-default tag-pill"
    data-tagname="training"
    href=""
  >
    training
  </a>
  <a
    className="tag-default tag-pill"
    data-tagname="tags"
    href=""
  >
    tags
  </a>
  <a
    className="tag-default tag-pill"
    data-tagname="as"
    href=""
  >
    as
  </a>
  <a
    className="tag-default tag-pill"
    data-tagname="coffee"
    href=""
  >
    coffee
  </a>
  <a
    className="tag-default tag-pill"
    data-tagname="animation"
    href=""
  >
    animation
  </a>
  <a
    className="tag-default tag-pill"
    data-tagname="flowers"
    href=""
  >
    flowers
  </a>
  <a
    className="tag-default tag-pill"
    data-tagname="caramel"
    href=""
  >
    caramel
  </a>
  <a
    className="tag-default tag-pill"
    data-tagname="money"
    href=""
  >
    money
  </a>
  <a
    className="tag-default tag-pill"
    data-tagname="cars"
    href=""
  >
    cars
  </a>
  <a
    className="tag-default tag-pill"
    data-tagname="baby"
    href=""
  >
    baby
  </a>
  <a
    className="tag-default tag-pill"
    data-tagname="happiness"
    href=""
  >
    happiness
  </a>
  <a
    className="tag-default tag-pill"
    data-tagname="sushi"
    href=""
  >
    sushi
  </a>
  <a
    className="tag-default tag-pill"
    data-tagname="sugar"
    href=""
  >
    sugar
  </a>
  <a
    className="tag-default tag-pill"
    data-tagname="clean"
    href=""
  >
    clean
  </a>
  <a
    className="tag-default tag-pill"
    data-tagname="japan"
    href=""
  >
    japan
  </a>
  <a
    className="tag-default tag-pill"
    data-tagname="cookies"
    href=""
  >
    cookies
  </a>
  <a
    className="tag-default tag-pill"
    data-tagname="well"
    href=""
  >
    well
  </a>
</div>
`;

exports[`>> Tags component snapshot returns null in case when tags is not exist 1`] = `
<div>
  Loading Tags...
</div>
`;
